core/chains/evm/log/helpers_test.go

```diff
@@ -387,20 +419,25 @@ type mockEthClientExpectedCalls struct {
 	FilterLogsResult []types.Log
 }
 
-func newMockEthClient(t *testing.T, chchRawLogs chan<- chan<- types.Log, blockHeight int64, expectedCalls mockEthClientExpectedCalls) *mockEth {
-	ethClient, sub := cltest.NewEthClientAndSubMock(t)
+func newMockEthClient(t *testing.T, chchRawLogs chan<- rawSub, blockHeight int64, expectedCalls mockEthClientExpectedCalls) *mockEth {
+	ethClient := new(evmmocks.Client)
+	ethClient.Test(t)
 	mockEth := &mockEth{
 		ethClient:       ethClient,
-		sub:             sub,
 		checkFilterLogs: nil,
 	}
 	mockEth.ethClient.On("ChainID", mock.Anything).Return(&cltest.FixtureChainID)
 	mockEth.ethClient.On("SubscribeFilterLogs", mock.Anything, mock.Anything, mock.Anything).
-		Run(func(args mock.Arguments) {
-			mockEth.subscribeCalls.Inc()
-			chchRawLogs <- args.Get(2).(chan<- types.Log)
-		}).
-		Return(mockEth.sub, nil).
+		Return(
+			func(ctx context.Context, q ethereum.FilterQuery, ch chan<- types.Log) ethereum.Subscription {
+				sub := mockEth.newSub(t)
+				chchRawLogs <- rawSub{ch, sub.Err()}
+				return sub
+			},
+			func(ctx context.Context, q ethereum.FilterQuery, ch chan<- types.Log) error {
+				return nil
+			},
```

Comment:
Mockery handles multi-return from actual args in an odd way....but this addresses a common frustration point at least :shrug: